app/src/main/java/tech/relaycorp/cogrpc/server/ClientsConnectedInterceptor.kt

```diff
@@ -0,0 +1,63 @@
+package tech.relaycorp.cogrpc.server
+
+import io.grpc.Grpc
+import io.grpc.Metadata
+import io.grpc.ServerCall
+import io.grpc.ServerCallHandler
+import io.grpc.ServerInterceptor
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.sendBlocking
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.launch
+import tech.relaycorp.courier.common.BehaviorChannel
+import tech.relaycorp.courier.common.Logging.logger
+import java.util.Date
+import kotlin.time.seconds
+
+/*
+ Counts the number of clients using the server.
+ Since each client will do multiple calls, that may be separated by intervals of 1 or 2 seconds,
+ we'll throttle the count number to keep it stable.
+ */
+class ClientsConnectedInterceptor(
+    private val coroutineScope: CoroutineScope
+) : ServerInterceptor {
```

Comment:
Did you consider using a `ServerTransportFilter` instead of a `ServerInterceptor`? That should allow us to have a simpler implementation as we could simply keep a thread-safe counter that's incremented or decremented as connections are opened or closed, respectively. See: https://stackoverflow.com/questions/41311530/grpc-number-of-active-clients-channels

I'm concerned that with the current approach depends on calls being completed in X seconds, which isn't very reliable.

Reply:
The server transport filter worked like a charm, it carries well across calls, so the clients number doesn't fluctuate.